CMakeLists.txt

```diff
@@ -135,9 +135,30 @@ set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${VERSION_MAJOR})
 
 set(CMAKE_C_FLAGS_DEBUGOPT "")
 
-target_compile_options(${PROJECT_NAME} PRIVATE -pedantic -std=gnu99 -Wall -Wimplicit -Wunused -Wcomment -Wchar-subscripts
-        -Wuninitialized -Wshadow -Wcast-align -Wwrite-strings -Wno-deprecated-declarations -Wno-unknown-pragmas -Wformat-security
-        -Wno-missing-braces -Wsign-compare -Wno-strict-prototypes -Wa,--noexecstack
+target_compile_options(${PROJECT_NAME} PRIVATE 
+    -pedantic
+    -std=gnu99
+    -Wall
+    -Wcast-align
+    -Wchar-subscripts
+    -Wcomment
+    -Wformat-security
+    -Wimplicit
+    -Wshadow
+    -Wsign-compare
+    -Wuninitialized
+    -Wunused
+    -Wwrite-strings
+    
+    # Assembler Options
+    -Wa,--noexecstack
+
+    # Suppressed Warnings
+    -Wno-deprecated-declarations
+    # GCC 4 fails to parse our macros with a "missing-braces" error
+    -Wno-missing-braces
```

Comment:
Should we have an issue for this? ARE macros missing braces?

Reply:
> ARE macros missing braces?

My understanding is that the answer is "no" otherwise things would actually be broken.

```
/codebuild/output/src491586333/src/github.com/aws/s2n-tls/s2n_head/tests/unit/s2n_x509_validator_time_verification_test.c: In function 'main':
/codebuild/output/src491586333/src/github.com/aws/s2n-tls/s2n_head/tests/unit/s2n_x509_validator_time_verification_test.c:116:34: warning: missing braces around initializer [-Wmissing-braces]
             DEFER_CLEANUP(struct s2n_stuffer cert_chain_stuffer = { 0 }, s2n_stuffer_free);
                                  ^
/codebuild/output/src491586333/src/github.com/aws/s2n-tls/s2n_head/./utils/s2n_safety.h:62:43: note: in definition of macro 'DEFER_CLEANUP'
     __attribute__((cleanup(_thecleanup))) _thealloc
                                           ^
```
[console link](https://us-west-2.console.aws.amazon.com/codesuite/codebuild/024603541914/projects/S2nIntegrationV2SmallBatch/build/S2nIntegrationV2SmallBatch%3A3b4bbc5e-6297-4c0b-bd26-7df3c51a95f7?region=us-west-2)

Other people have run into similar false positives with old versions of GCC.
https://github.com/OP-TEE/optee_os/issues/1998

I did find a [bug report](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53119) that seems related, but it says the fix was backported to GCC 4.8.3 and our CI version is 4.8.5

Reply:
https://godbolt.org/z/qEW1sePYs

Okay our macros are NOT missing braces. We're not going to zero-init like that xD